File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/EntityUtils.java
package org.metabrainz.mobile.data.sources.api.entities;

import java.util.Iterator;
import java.util.List;


public class EntityUtils {

    public static String getDisplayArtist(List<ArtistCredit> artistCredits) {
        StringBuilder builder = new StringBuilder();
        Iterator<ArtistCredit> iterator = artistCredits.iterator();
        while (iterator.hasNext()) {
            ArtistCredit credit = iterator.next();
            if (credit != null && credit.getName() != null &&
                    !credit.getName().equalsIgnoreCase("null")) {
                builder.append(credit.getName());
                if (iterator.hasNext())
                    builder.append(credit.getJoinphrase());
            }
        }
        return builder.toString();
    }

}

File: app/src/main/java/org/metabrainz/android/data/sources/api/entities/EntityUtils.kt
package org.metabrainz.android.data.sources.api.entities

object EntityUtils {
    @JvmStatic
    fun getDisplayArtist(artistCredits: MutableList<ArtistCredit>?): String {
        val builder = StringBuilder()
        val iterator = artistCredits!!.iterator()
        while (iterator.hasNext()) {
            val credit = iterator.next()
            if (credit.name != null &&
                    !credit.name.equals("null", ignoreCase = true)) {
                builder.append(credit.name)
                if (iterator.hasNext()) builder.append(credit.joinphrase)
            }
        }
        return builder.toString()
    }
}